cmd/promql-compliance-tester/main.go

```diff
@@ -15,17 +16,28 @@ import (
 	"github.com/promlabs/promql-compliance-tester/testcases"
 )
 
-func newPromAPI(url string) (v1.API, error) {
-	client, err := api.NewClient(api.Config{
-		Address: url,
-	})
+func newPromAPI(targetConfig config.TargetConfig) (v1.API, error) {
+	apiConfig := api.Config{Address: targetConfig.QueryURL}
+	if len(targetConfig.XQueryKey) > 0 {
```

Comment:
I know others will also want to set custom headers, so let's just allow configuring an arbitrary `map[string]string` vs. hardcoding a particular header name :)

Reply:
I started down that path, and went with the specific header because it was easier (and I don't know Go at all).  But I agree. I'll update this, but probably not for at least a few workdays.